src/deploy/functions/params.ts

```diff
@@ -0,0 +1,332 @@
+import { logger } from "../../logger";
+import { FirebaseError } from "../../error";
+import { promptOnce } from "../../prompt";
+import * as build from "./build";
+import { assertExhaustive } from "../../functional";
+
+type CEL = build.Expression<string> | build.Expression<number> | build.Expression<boolean>;
+function isCEL(expr: string | number | boolean): expr is CEL {
+  return typeof expr === "string" && expr.includes("{{") && expr.includes("}}");
+}
+function dependenciesCEL(expr: CEL): string[] {
+  return /{{ params\.\w+ }}/.exec(expr)?.slice(1) || [];
+}
+
+/**
+ * Resolves a numeric field in a Build to an an actual numeric value.
+ * Fields can be literal or a {{ }} delimited CEL expression referencing param values.
+ * Currently only the CEL identity {{ params.PARAMNAME }} is implemented.
+ */
+export function resolveInt(
+  from: number | build.Expression<number>,
+  paramValues: Record<string, build.Field<string | number | boolean>>
+): number {
+  if (typeof from === "number") {
+    return from;
+  }
+  if (!/{{ params\.(\w+) }}/.test(from)) {
```

Comment:
just do an `exec` call once and test whether the result is truthy or not. It'll also save you from doing the null assertion.